File: services/api_gateway/src/main/java/com/cvassistant/apigateway/model/CVHistory.java
package com.cvassistant.apigateway.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import java.time.LocalDateTime;

@Entity
@Table(name = "cv_history")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class CVHistory {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    private String fileName;
    private String fileUrl;
    
    @Column(columnDefinition = "TEXT")
    private String extractionResult;
    
    private LocalDateTime uploadedAt;

    @PrePersist
    protected void onUpload() {
        uploadedAt = LocalDateTime.now();
    }
}

File: services/api_gateway/CvAssistant.ApiGateway.Domain/Entities/CVHistory.cs
namespace CvAssistant.ApiGateway.Domain.Entities;

public class CVHistory
{
    public long Id { get; set; }
    public long UserId { get; set; }
    public string? FileName { get; set; }
    public string? FileUrl { get; set; }
    public string? ExtractionResult { get; set; }
    public DateTime UploadedAt { get; set; }

    // Navigation property
    public User? User { get; set; }
}
